### ISSUE TITLE
Ability to migrate accounts bеtween domains

### ISSUE THREAD

**Issue by top-secrett** [apache/cloudstack#7555]
##### ISSUE TYPE

 * Feature Idea

##### COMPONENT NAME

~~~
UI
~~~

##### SUMMARY

It would be great if CS has function to migrate accounts and its resourses between different domains. 
We created stucture of domains based on some logic. Now we want to change it, but can't do it.

**Comment by top-secrett** [comment]
@DaanHoogland yeah, you're right. For example, from migrate account form domain VS/subdomain1 to VS/121/subdomain1

**Comment by DaanHoogland** [comment]
ok, so not move domain sub1 from /dom1/sub1 to dom2/sub1? 
in any case that may be a different feature and a first step just the account.
thanks

**Comment by top-secrett** [comment]
@DaanHoogland I think account moving is more flexible at least in our case. But domain moving (with all accounts inside?) would be great too.

**Comment by DaanHoogland** [comment]
@top-secrett initial investigation leads to several issues. A non-exhaustive list:

1. an account does not have to be unique globally but has to be unique within a domain
2. usage data for a resource contains account and domain data that will not be consistent anymore after moving an account

that said, it is a good feature and worth investigating. cc @vishesh92 @rohityadavcloud

**Comment by DaanHoogland** [comment]
> > @DaanHoogland I think account moving is more flexible at least in our case. But domain moving (with all accounts inside?) would be great too.
> 
> @top-secrett PR #7446 adds the feature to move subdomains between domains (with everything that inside them). As @DaanHoogland said, account migration is still a different feature from #7446; however, PR #7446 may be a first step towards account migration.

you are right @JoaoJandre , domain moving is actually easier as it removes the necessity to check for uniqueness of account names. Also usage records remain valid as the account-domain relation stays the same. Let's get your #7446 in before we considder this.

**Comment by DaanHoogland** [comment]
@JoaoJandre there is already an implementation of moveDomain by now as you implemented in #7446 , should we still keep this open? (moveAccount)

my objection (.1) from https://github.com/apache/cloudstack/issues/7555#issuecomment-1564341238 can easily be mitigated by throwing some NotAllowedException, but objection (.2), histroic usage remains, So I tend towards closing this  as unplanned.

**Comment by JoaoJandre** [comment]
> [@JoaoJandre](https://github.com/JoaoJandre) there is already an implementation of moveDomain by now as you implemented in [#7446](https://github.com/apache/cloudstack/pull/7446) , should we still keep this open? (moveAccount)
> 
> my objection (.1) from [#7555 (comment)](https://github.com/apache/cloudstack/issues/7555#issuecomment-1564341238) can easily be mitigated by throwing some NotAllowedException, but objection (.2), histroic usage remains, So I tend towards closing this as unplanned.

@DaanHoogland as no one currently seems interested in implementing this. I would close it. It seems a bit too complex to implement and I do not see much gain from it.
